### DISCUSSION TITLE
Kinesis On-demand Mode

### DISCUSSION THREAD

**Discussion by Bin-security** [brexhq/substation#90] [Q&A] [answered]
Hey @jshlbrd Maybe you have considered this or Substation supports it already. It seems that Kinesis supports this [On-demand mode](https://docs.aws.amazon.com/streams/latest/dev/how-do-i-size-a-stream.html) nowadays. It can scale up and down automatically to support gigabytes of write and read per minute. Have you looked at the Kinesis on-demand mode for substation? That would simplify the design a little as it currently uses a lambda to autoscale up and down the number of shards.

**Comment by jshlbrd** [comment, marked as answer]
Hey @Bin-security 👋 

Substation already supports Kinesis Data Streams' On-Demand capacity, it's [configurable with Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/kinesis_stream#stream_mode_details) (but obviously not compatible with the custom autoscaler).

We've looked at On-Demand in addition to other streaming configurations (Firehose and AWS MSK) and determined that the Provisioned capacity with our autoscaler is the most affordable solution. By our estimates, On-Demand costs between 15x and 20x the cost of Provisioned with our autoscaler.
